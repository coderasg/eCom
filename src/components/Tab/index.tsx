import React from 'react';
import ArrowDown from './../Icons/ArrowDown/index.svg';
import {$topPageBackgroundColor } from './../Icons/IconColors';

import styles from './tab.module.scss';

type Props = {
  name: string;
  isSelected: boolean;
  handleTabClick: () => void;
};

function Tab(props: Props) {
  const { name, handleTabClick, isSelected } = props;

  return (
    <div
      data-testid="Tab-wrapper"
      onClick={handleTabClick}
      className={`${styles.tab} ${isSelected && styles.selectedTab}`}
    >
      {name}
      {isSelected && <ArrowDown width={20} height={20} className={styles.arrowDown} fill={$topPageBackgroundColor} />}
    </div>
  );
}

export default Tab;
