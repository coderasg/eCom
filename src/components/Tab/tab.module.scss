@use "./../../variables";

.tab {
  padding: 2px;
  margin-right: 10px;
  cursor: pointer;
  white-space: nowrap;
}

.selectedTab {
  color: red;
  position: relative;

  .arrowDown {
    position: absolute;
    left: 0;
    top: 23px;
    width: 100%;
    transform: rotate(180deg);
  }
}
