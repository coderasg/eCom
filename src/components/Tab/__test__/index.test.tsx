import React from 'react';
import renderer from 'react-test-renderer';

import Tab from '../index';

describe('(Component) Tab', () => {
  const defaultProps = {
    name: 'test',
    isSelected: true,
    handleTabClick: () => {
      alert('tab');
    },
  };

  test('should pass snapshot test', () => {
    const wrapper = renderer.create(<Tab {...defaultProps} />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
