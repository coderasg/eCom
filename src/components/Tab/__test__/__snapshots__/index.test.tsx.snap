// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Tab should pass snapshot test 1`] = `
<div
  className="undefined undefined"
  data-testid="Tab-wrapper"
  onClick={[Function]}
>
  test
  <SvgIcon
    fill="#f1f1f1"
    height={20}
    width={20}
  />
</div>
`;
